### PR TITLE
Fix warnings related to React@15.5.x

### PR DESCRIPTION
**Pull request by samhogg**
This PR fixes all console spam when upgrading to React 15.5.x, apart from a propTypes warning from deck.gl. I've put together a PR for, that too.

### CODE REVIEW

**Comment by samhogg** [comment]
Hmmmm, not sure what regressed there, looking into it.

**Comment by samhogg** [comment]
Build is magically passing, wooo

**Comment by samhogg** [comment]
Any objection to getting this merged, @mcnuttandrew @javidhsueh?

File: docs/package.json [hunk 1]
```diff
@@ -5,9 +5,9 @@
     "d3-request": "^1.0.5",
     "immutable": "^4.0.0-rc.1",
     "markdown-loader": "^2.0.0",
-    "react": "^15.4.2",
-    "react-addons-shallow-compare": "^15.4.2",
-    "react-dom": "^15.4.2",
+    "react": "^15.5.4",
+    "react-addons-shallow-compare": "^15.5.2",
```
Comment (chrisirhc) [inline docs/package.json]: Looks like this is deprecated. Will add an issue for task to refactor into using `React.PureComponent`.

**Comment by chrisirhc** [comment]
No objection, looks good.

File: docs/package.json (hunk 1 above)
Comment (samhogg) [inline docs/package.json]: Good call

**Comment by mcnuttandrew** [comment]
Can someone @chrisirhc, @btford, @javidhsueh, merge this?

**Pull request merged**